07-errors.md

```diff
@@ -0,0 +1,301 @@
+| 10    | `SysErrAssertionFailed`    | some internal assertion failed                            |
+| 11-15 | reserved                   |                                                           |
+
+Examples:
+
+1. If a message sender doesn't exist on-chain, the VM will return a receipt with the
+   `ExitCode::SysErrSenderInvalid` abnormal exit code.
+2. If an actor calls `vm::abort` with a system abnormal exit code, the send to that actor
+   will fail with the `ExitCode::ErrIllegalActor` _syscall error_. If this is a top-level send (the
+   execution of an on-chain message), the `ExitCode::SysErrIllegalActor` system exit code will be
+   recorded as the messages exit code.
+3. If a message runs out of gas, the VM will return a receipt with the `ExitCode::SysErrOutOfGas`
+   abnormal exit code.
+
+Changes from pre-FVM Filecoin:
+
+1. Removed `ExitCode::SysErrInvalidMethod` (3). Dispatch is now handled inside actor code.
+2. Removed `ExitCode::SysErrForbidden` (8). Caller verification is a userspace operation, we'll now
+   exit with `ExitCode::ErrForbidden`.
+4. Replaced `ExitCode::SysErrIllegalArgument` with `ExitCode::SysErrAssertionFailed` to more
+   accurately reflect its usage. When this appears on-chain, it means that a message hit some
+   internal Filecoin assert that shouldn't happen.
+
+### Standard "Abnormal Exit" Codes
+
+Standard abnormal exit codes are codes returned by actors to indicate common abnormal exit conditions.
+
+| Code  | Status                 | Description                                         |
+|-------|------------------------|-----------------------------------------------------|
+| 16    | `ErrIllegalArgument`   | invalid message parameters                          |
+| 17    | `ErrNotFound`          | message referenced something that doesn't exist     |
+| 18    | `ErrForbidden`         | operation forbidden                                 |
+| 19    | `ErrInsufficientFunds` | insufficient funds for operation                    |
+| 20    | `ErrIllegalState`      | the actor is in an illegal state                    |
+| 21    | `ErrSerialization`     | actor failed to serialize or deserialize some state |
+| 22    | `ErrInvalidMethod`     | specified method does not exist                     |
+| 23    | `ErrUnspecified`       | the actor failed with an unspecified error          |
+| 24-31 | reserved               |                                                     |
+
+Changes from pre-FVM Filecoin:
+
+1. Added `ExitCode::ErrInvalidMethod` (replacing `ExitCode::SysErrInvalidMethod`).
+2. Added `ExitCode::ErrUnspecified`. This generally means "there's a bug and I have no idea what
+   went wrong". The alternative would be for the actor to _panic_ (leading to an
+   `ExitCode::ErrIllegalActor` syscall error), but it may be useful to distinguish between
+   "crashing" and "something unknown went wrong".
+
+## Syscall Error Numbers
```

Comment:
This way, this term is not at odds with 0.

```suggestion
## Syscall Return Codes
```

Reply:
1. I'm trying to avoid reusing "code".
2. In the SDK, there will be an `ErrorNumber` enum and "0" won't be represented. I expect this'll be the case in most languages.

So I'd _really_ rather just talk about this in terms of error numbers, leaving "0" as a "this isn't an error" status code.